### PR TITLE
updated for notification: model, controller and routes

### PR DESCRIPTION
**Pull request by christian-barnabe-chabi**
### What this PR does (required):
- updates for notification: model, controller and routes to match the needed data in FE notification template

### Screenshots / Videos (required):
- Post at least one screenshot to show the feature in action

### Any information needed to test this feature (required):
- Post what steps to take to be able to test this feature (eg. "sign up as a new user and upload a profile picture from the navbar")

### Any issues with the current functionality (optional):
- List any problems where you were not able to complete all tasks on ticket
- List any problems this PR may cause for the project or other active PRs
- Post a screenshot/video of the problem, along with detailed console outputs where applicable

### CODE REVIEW

File: server/controllers/notification.js [hunk 1]
```diff
@@ -58,30 +90,33 @@ exports.markAsread = expressAsyncHandler(async (req, res, next) => {
...
-          message: error,
-        }
-      });
-    } else {
-      res.status(201).json({
-        success: {
-          message: "Notification updated",
-        }
-      });
+  await Notification.updateOne(
```
Comment (dproc96) [inline server/controllers/notification.js:100]: If you've already used find to get the notification, you can just update directly and save it, you don't need to use updateOne

File: server/models/Notification.js [hunk 2]
```diff
@@ -14,17 +14,20 @@ const notificationSchema = new mongoose.Schema({
...
+    required: false,
+  },
+  url: {
+    type: String,
     required: false,
-    ref: 'products',
   },
   old_price: {
-    type: Number,
+    type: String,
```
Comment (dproc96) [inline server/models/Notification.js:26]: Why did we update to String here?

**Review by dproc96 — COMMENTED** [review]
Had two small thoughts about this, just want to clarify the changes on the model before approving

File: server/models/Notification.js (hunk 2 above)
Comment (christian-barnabe-chabi) [reply server/models/Notification.js:26]: we do not yet handle currencies

File: server/controllers/notification.js [hunk 3]
```diff
@@ -35,20 +35,52 @@ exports.getUnread = expressAsyncHandler(async (req, res, next) => {
...
+exports.markAllAsRead = expressAsyncHandler(async (req, res, next) => {
   const user = await User.findById(req.user.id);
 
   if (!user) {
     res.status(401).send("Unathenticated");
     throw new Error("Not authorized");
   }
 
+  notification.read = true;
+  notification.read_date = new Date().toString();
```
Comment (moffatethan) [inline server/controllers/notification.js:51]: What is the plan with read_date? For filtering notifcations by when they read them?

Comment (moffatethan) [reply server/controllers/notification.js:51]: Also, please use camelCase

File: server/models/Notification.js [hunk 4]
```diff
@@ -14,17 +14,20 @@ const notificationSchema = new mongoose.Schema({
...
+  image: {
+    type: String,
+    required: false,
+  },
+  url: {
+    type: String,
     required: false,
-    ref: 'products',
   },
   old_price: {
```
Comment (moffatethan) [inline server/models/Notification.js:25]: JavaScript convention is camelCase and not snake_case

File: server/models/Notification.js [hunk 5]
```diff
@@ -14,17 +14,20 @@ const notificationSchema = new mongoose.Schema({
...
-    type: Number,
+    type: String,
     required: false,
   },
   new_price: {
```
Comment (moffatethan) [inline server/models/Notification.js:29]: Same here, camelCase

**Review by moffatethan — APPROVED** [review]
Looks good to me, left some comments to address before merging.

**Comment by christian-barnabe-chabi** [comment]
> Looks good to me, left some comments to address before merging.

All suggestions applied

**Pull request merged**